src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/OutOfWorkspaceCompletion.test.ts

```diff
@@ -0,0 +1,49 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from 'assert';
+import { afterEach, before, beforeEach } from 'mocha';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import {
+    mvcWithComponentsRoot,
+    pollUntil,
+    testAppsRoot,
+    waitForProjectReady,
+} from './TestUtil';
+
+suite('Completions', () => {
+    before(async () => {
+        await waitForProjectReady(mvcWithComponentsRoot);
+    });
+
+    afterEach(async () => {
+        await vscode.commands.executeCommand('workbench.action.revertAndCloseActiveEditor');
+        await pollUntil(async () => {
+            await vscode.commands.executeCommand('workbench.action.closeAllEditors');
+            if (vscode.window.visibleTextEditors.length === 0) {
+                return true;
+            }
+
+            return false;
+        }, /* timeout */ 3000, /* pollInterval */ 500, true /* suppress timeout */);
+    });
+
+    test('Completions out of Workspace work', async () => {
```

Comment:
It'd probably be nice to expand this test to verify directives and C# (I believe DateTime will be available even for non-workspace files)